### PR TITLE
126 fix donation handling

### PR DESCRIPTION
**Pull request by aliberkhsl**
#126

### CODE REVIEW

File: contracts/staking/BaseShareField.sol [hunk 1]
```diff
@@ -253,9 +272,8 @@ contract BaseShareField is DSMath {
 		UserInfo storage userInfo = users[user];
 		_canMintRewards();
 		_update(rewardsPerBlock, blockStart, blockEnd);
-		_audit(user, userInfo.amount);
+		_audit(user, userInfo.amount, 1);
```
Comment (sirpy) [inline contracts/staking/BaseShareField.sol:275]: 1?

**Review by sirpy — CHANGES_REQUESTED** [review]
add tests to check  all new variables, effectivestake etc...
add tests to check their calculations no only withdrawing proportionaly

File: contracts/staking/BaseShareField.sol [hunk 2]
```diff
@@ -106,13 +112,25 @@ contract BaseShareField is DSMath {
...
+			if (_amount > 0) {
+				uint256 withdrawFromEffectiveStake =
+					((_amount - updatedAmount) * userInfo.effectiveStakes) /
+						_amount;
+				userInfo.effectiveStakes -= withdrawFromEffectiveStake;
+				totalEffectiveStakes -= withdrawFromEffectiveStake;
+			}
+		}
 
+		if (_donationPer == 0) {
```
Comment (sirpy) [inline contracts/staking/BaseShareField.sol:126]:
```suggestion
		else if (_donationPer == 0) {
```

File: contracts/staking/BaseShareField.sol (hunk 1 above)
Comment (aliberkhsl) [reply contracts/staking/BaseShareField.sol:275]: Doesn't matter just needs to be something different than 0

Comment (sirpy) [reply contracts/staking/BaseShareField.sol:275]: yes  i know, i can read the code.
what do we do when we write something that isnt obvious??

Comment (aliberkhsl) [reply contracts/staking/BaseShareField.sol:275]: Added comment

**Pull request merged**